tests/behat/features/bootstrap/EmailContext.php

```diff
@@ -18,11 +20,27 @@ class EmailContext implements Context {
    * @BeforeScenario @email-spool
    */
   public function enableEmailSpool() {
-    // Update Drupal configuration.
-    $swiftmailer_config = \Drupal::configFactory()->getEditable('swiftmailer.transport');
-    $swiftmailer_config->set('transport', 'spool');
-    $swiftmailer_config->set('spool_directory', $this->getSpoolDir());
-    $swiftmailer_config->save();
+    if ($default = MailerTransport::loadDefault()) {
+      \Drupal::state()->set('default_mailer_transport', $default->id());
+    }
+
+    if (!$entity = MailerTransport::load('behat_file_spool')) {
+      $entity = MailerTransport::create([
+        'id' => 'behat_file_spool',
+        'label' => 'Behat File Spool',
+        'plugin' => 'null',
+      ]);
+      $entity->save();
+    }
+
+    $entity->setAsDefault();
+
+    // Add the mail logger plugin to the default policy.
+    $default_mailer_policy = MailerPolicy::load('_');
+    $mailer_configuration = $default_mailer_policy->getConfiguration();
+    $mailer_configuration['log_mail'] = ['spool_directory' => $this->getSpoolDir()];
+    $default_mailer_policy->setConfiguration($mailer_configuration);
+    $default_mailer_policy->save();
```

Comment:
We're using the Entity system here which might cause conflicts between the test runner and the actual database.

The Behat test runner has a local bootstrap of Drupal in-memory. This will have statically cached entities, that [should be automatically reset](https://github.com/goalgorilla/open_social/blob/main/tests/behat/features/bootstrap/DatabaseContext.php#L180). However it's possible that the entity is in memory for the Behat runner which causes `MailerTransport::load` to return something, while the entity is not actually in the database anymore (which was freshly loaded from the dump).

In that case the transport won't be created and a request to Drupal will run without it being configured.

Reply:
changed to use the config factory